include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -123,38 +125,58 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
 
     // TODO: filter hits and only do it when necessary (depending on error types)
 
+    constexpr bool output_cursor = cfg_t::template exists<search_cfg::output<detail::search_output_index_cursor>>();
```

Comment:
I have started to make a global traits type with all necessary types and constants defined based on the configuration. This way you don't need to expose this cumbersome `template exists` syntax. I found this a little easier to read and to use throughout the entire code base